### PR TITLE
Update CONTRIBUTING.md for default branch

### PR DESCRIPTION
**Pull request by backwind1233**
As title

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
 
 1. Make your changes in a new git fork.
-1. Checkout a new  feature/bugfix branch from `main`.
-2. Develop in feature/bugfix branch.
-3. Make pull request to merge the branch into main.(able to merge)
-
-
-### 02. Develop sample with released versions of libraries(not the latest version)
+2. Checkout a new  feature/bugfix branch from `main`.
+    > The `main` branch is a develop branch, the samples may depend on beta version libraries, which are not released to maven central.
```
Comment (jialigit) [inline CONTRIBUTING.md:8]: How about we use stable and released version for this main branch. If the user wants to test the latest code, just tell them to install the dependency from source and change the pom themselves.

For developing 4.1.x, we create corresponding branch for sample.So,
main ---> depends on the latest stable released lib. No need to install dependencies.
4.1.x ---> depends on 4.1.x  of unreleased lib, users have to install using the source code.
after 4.1.x released, merge to main branch

Comment (backwind1233) [reply CONTRIBUTING.md:8]: @jialigit we keep the same branch policy for SDK, Samples and reference docs, so main branch is the develop branch,
but we set the default branch of samples with the latest relased version.

Comment (saragluna) [reply CONTRIBUTING.md:8]: We discussed this before and concluded on the principle of `developing the latest changes using the main branch`, which will make the behavior consistent across our three repos: sample-repo, javadoc-repo, source-code repo.

**Review by saragluna — APPROVED** [review]
LGTM.

**Review by yiliuTo — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
-1. Make your changes in a new git fork.
-1. Checkout a new feature/bugfix branch.
-2. Specify the version of libraries.
 3. Develop in feature/bugfix branch.
-3. Make pull request to merge the branch into new **feature** branch.
-
-### 03. Develop sample with unreleased libraries
-
-In this scenario, the samples may depend on beta version library, which is not released to maven central.
+4. Build the unreleased dependency libraries in your local machine.
```
Comment (chenrujun) [inline CONTRIBUTING.md:11]: Why `unrelated`?

**Review by chenrujun — APPROVED** [review]

**Pull request merged**